geopandas/io/file.py

```diff
@@ -582,10 +582,12 @@ def _to_file_pyogrio(df, filename, driver, schema, crs, mode, **kwargs):
             "The 'schema' argument is not supported with the 'pyogrio' engine."
         )
 
-    if mode != "w":
-        raise ValueError(
-            "Only mode='w' is supported for now with the 'pyogrio' engine."
-        )
+    if mode not in ("w", "a"):
+        raise ValueError("'mode' should be one of 'w' or 'a', got '{mode}' instead")
+
+    if "append" not in kwargs:
```

Comment:
I'm not sure why someone would attempt to do so, but it seems possible that someone could call this with `mode="a", append=False`, which would then not produce the expected behavior (overwrite instead of append).  Since append is not the default, we can probably get away with 

```Python
if mode == "a":
    kwargs["append"] = True
```